addons/web/static/src/search/control_panel/control_panel.js

```diff
@@ -233,4 +335,204 @@ export class ControlPanel extends Component {
             ev.stopPropagation();
         }
     }
+
+    /**
+     * @param {TopBarAction} action
+     */
+    _checkValueLocalStorage(action) {
+        const actionIdStr = action.id.toString();
+        return this.state.visibleTopBarActions[actionIdStr];
+    }
+
+    /**
+     * The selected action is put into (or removed from) the localStorage and its visibility changes.
+     * The state variable visibleTopBarActions keeps track of the visible actions to avoid  having to parse
+     * the localStorage values every time we want to access them.
+     * @param {TopBarAction} action
+     */
+    _setVisibility(action) {
+        const actionIdStr = action.id.toString();
+        if (this.state.visibleTopBarActions[actionIdStr]) {
+            delete this.state.visibleTopBarActions[actionIdStr];
+        } else {
+            this.state.visibleTopBarActions[actionIdStr] = true;
+        }
+        localStorage.setItem(
+            this.localStorageKeyActionIdResId,
+            JSON.stringify(this.state.visibleTopBarActions)
+        );
+    }
+
+    _onShareCheckboxChange() {
+        this.state.newActionShared = !this.state.newActionShared;
```

Comment:
I think that is the whole purpose of a controlled state. Being able to control and read the state instead of having to get the value of the checkbox in the dom